stac_validator/stac_validator.py

```diff
@@ -75,86 +74,59 @@ def __init__(self, stac_file, version="master", loglevel="CRITICAL"):
             "unknown": 0,
         }
 
-    def fetch_specs(self, version):
+    @lru_cache(maxsize=48)
+    def fetch_spec(self, spec):
         """
-        Get the versions from github. Cache them if possible.
-        :return: specs
+        Get the spec file and cache it.
+        :param spec: name of spec to get
+        :return: STAC spec in json format
         """
-        geojson_key = "geojson_resolver"
-        item_key = f"item-{self.stac_version}"
-        catalog_key = f"catalog-{self.stac_version}"
-
-        if item_key in cache and catalog_key in cache:
-            logging.debug("Using STAC specs from local cache.")
-            self.geojson_resolver = RefResolver(
-                base_uri=f"file://{self.dirpath}/", referrer="geojson.json"
-            )
-            return cache[item_key], cache[geojson_key], cache[catalog_key]
 
+        if spec == "geojson":
+            spec_name = "geojson"
+        elif spec == "catalog" or spec == "collection":
+            spec_name = "catalog"
         else:
-            try:
-                logging.debug("Gathering STAC specs from remote.")
-                stac_item_geojson = requests.get(
-                    StacVersion.item_geojson_schema_url(version)
-                ).json()
-                stac_item = requests.get(StacVersion.item_schema_url(version)).json()
-                stac_catalog = requests.get(
-                    StacVersion.catalog_schema_url(version)
-                ).json()
-            except Exception as error:
-                logger.exception("STAC Download Error")
-                raise VersionException(
-                    f"Could not download STAC specification files for version: {version}"
-                )
+            spec_name = "item"
```

Comment:
This might be a good place to use a case statement, instead?

Reply:
python doesn't have case statements. I supposed we could do something with dictionaries for when we add extensions.